bundles/io.openliberty.tools.eclipse.lsp4e/src/io/openliberty/tools/eclipse/liberty/languageserver/LibertyLSClientImpl.java

```diff
@@ -0,0 +1,108 @@
+import org.eclipse.lsp4e.LanguageClientImpl;
+import org.eclipse.lsp4j.DidChangeWatchedFilesParams;
+import org.eclipse.lsp4j.FileChangeType;
+import org.eclipse.lsp4j.FileEvent;
+import org.eclipse.swt.widgets.Display;
+
+import io.openliberty.tools.eclipse.ls.plugin.LibertyToolsLSPlugin;
+
+/**
+ * Liberty Config language client.
+ * 
+ * @author
+ */
+public class LibertyLSClientImpl extends LanguageClientImpl {
+
+    public LibertyLSClientImpl() {
+        super();
+        IWorkspace iWorkspace = ResourcesPlugin.getWorkspace();
+        LCLSListener resourceChangeListener = new LCLSListener();
+        iWorkspace.addResourceChangeListener(resourceChangeListener, IResourceChangeEvent.POST_BUILD);
+
+    }
+
+    public void fireUpdate(String uri) {
+        List<FileEvent> fileEvents = new ArrayList<FileEvent>();
+        fileEvents.add(new FileEvent(uri, FileChangeType.Changed));
+        DidChangeWatchedFilesParams params = new DidChangeWatchedFilesParams();
+        params.setChanges(fileEvents);
+
+        getLanguageServer().getWorkspaceService().didChangeWatchedFiles(params);
+    }
+
+    public class LCLSListener implements IResourceChangeListener {
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void resourceChanged(IResourceChangeEvent event) {
+            Display.getDefault().syncExec(new Runnable() {
+
+                @Override
+                public void run() {
+
+                    IResourceDelta delta = event.getDelta();
+                    if (delta == null) {
+                        return;
+                    }
+
+                    final ArrayList<String> changed = new ArrayList<String>();
+
+                    IResourceDeltaVisitor visitor = new IResourceDeltaVisitor() {
+                        public boolean visit(IResourceDelta delta) {
+                            IResource resource = delta.getResource();
+                            if (resource.getType() == IResource.FILE) {
+
+                                // Look for changes to liberty-plugin-config.xml, *.properties, and *.env
+                                if ("liberty-plugin-config.xml".equalsIgnoreCase(resource.getName())
+                                        || "properties".equalsIgnoreCase(resource.getFileExtension())
+                                        || "env".equalsIgnoreCase(resource.getFileExtension())) {
+                                    changed.add(resource.getLocationURI().toString());
+                                }
+                            }
+                            return true;
+                        }
+                    };
+
+                    try {
+                        delta.accept(visitor);
+                    } catch (CoreException e) {
+                        LibertyToolsLSPlugin.logException(e.getLocalizedMessage(), e);
+                    }
+
+                    for (String change : changed) {
+                        fireUpdate(change);
```

Comment:
Since the DidChangeWatchedFilesParams holds a List of changes would it be better to set them as a List and send as a single update?

Reply:
combined into a single update 👍